### PR TITLE
feat: real time progress updates / refactor

### PR DESCRIPTION
**Pull request by devinburnette**
Took a stab at a larger refactor for real time progress updates based on feedback here: https://github.com/stemrollerapp/stemroller/pull/45

Let me know if this is closer to what you had in mind.

### CODE REVIEW

**Comment by iffyloop** [comment]
Am on my phone currently so can't review it fully, but at a quick glance this is almost exactly what I had in mind. Thanks for your contribution! I'll work on getting it merged in next time I have a chance to focus on StemRoller.

File: main-src/processQueue.cjs [hunk 1]
```diff
@@ -207,7 +229,7 @@ async function _processVideo(video, tmpDir) {
   if (PATH_TO_MODELS) {
     demucsExeArgs.push('--repo', PATH_TO_MODELS)
   }
-  await spawnAndWait(tmpDir, DEMUCS_EXE_NAME, demucsExeArgs)
+  await spawnAndWait(video.videoId, tmpDir, DEMUCS_EXE_NAME, demucsExeArgs)
```
Comment (devinburnette) [inline main-src/processQueue.cjs:232]: this feels a little wonky having to pass the video ID all the way down like this, what do you think?

**Comment by blaise-tk** [comment]
any news on this update?

**Comment by iffyloop** [comment]
Sorry, have been really busy the past few weeks. Am still planning to merge this in, just haven't had the time to review and test on my local devices yet.

**Comment by iffyloop** [comment]
@devinburnette this is now merged into main, thanks again for adding this feature! Super helpful and I think will make an especially significant difference for new users, who aren't aware of how long StemRoller will take to process files.

**Comment by iffyloop** [comment]
@blaise-tk feature has been merged

**Pull request merged**